### PR TITLE
Generate lenses with hprotoc (version >= 2.1.7 required)

### PR DESCRIPTION
**Pull request by timjb**
instead of the "incredibly hacky" python script.

### CODE REVIEW

File: NN/Examples/GoogLeNet.hs [hunk 1]
```diff
@@ -48,14 +48,14 @@ inception input Inception{..} = do
        [googleConv & numOutputC' _1x1  & kernelSizeC' 1 & weightFillerC' (xavier 0.03), relu],
        [googleConv & numOutputC' _3x3reduce & kernelSizeC' 1 & weightFillerC' (xavier 0.09), relu, googleConv & numOutputC' _3x3 & kernelSizeC' 3 & weightFillerC' (xavier 0.03) & padC' 1, relu],
        [googleConv & numOutputC' _5x5reduce & kernelSizeC' 1 & weightFillerC' (xavier 0.2), relu, googleConv & numOutputC' _5x5 & kernelSizeC' 5 & weightFillerC' (xavier 0.03) & padC' 2, relu],
-       [maxPool& sizeP' 3 & strideP' 3 & padP' 1, googleConv & numOutputC' _poolProj & kernelSizeC' 1 & weightFillerC' (xavier 0.1), relu]]
+       [maxPool & sizeP' 3 & strideP' 3 & padP' 1, googleConv & numOutputC' _poolProj & kernelSizeC' 1 & weightFillerC' (xavier 0.1), relu]]
```
Comment (ajtulloch) [inline NN/Examples/GoogLeNet.hs]: Does _poolProj need to change?

**Comment by ajtulloch** [comment]
Looks great. Is it possible to add an hprotoc >= 2.1.7 dependency somewhere (stack.yaml?)

**Comment by timjb** [comment]
You could put `resolver: nightly-2015-09-21` in `stack.yaml`. This version of Stackage includes `hprotoc` version 2.1.7. You will then get this version when you run `stack install hprotoc` while in the project directory. You probably don't want to list `hprotoc` explicitly as a dependency in the cabal or stack files because it is only a development dependency, not one that is needed for compilation.

**Comment by ajtulloch** [comment]
Awesome, thanks so much.

**Pull request merged**